src/raven.js

```diff
@@ -5,6 +5,8 @@
 // since JSON is required to encode the payload
 var _Raven = window.Raven,
     hasJSON = !!(typeof JSON === 'object' && JSON.stringify),
+    // Raven can run in contexts where there's no document (react-native)
+    hasDocument = typeof document !== 'undefined',
```

Comment:
I know it does, but it seems gross to assume that a function is hoisted on the very JavaScript statement in a file.